web/client/hooks/__tests__/useModulePlugins-test.js

```diff
@@ -33,34 +33,130 @@ describe('useModulePlugins hook', () => {
+                muteEpics: () => {
+                    // we should not end up in here
+                    expect(true).toBe(false);
+                },
+                unmuteEpics: () => {},
+                rootEpic: () => {}
+            }
+        };
+        setStore(newStore);
+
+        const plugins = {
+            ExamplePlugin: toModulePlugin('Example', () => import('../../test-resources/module-plugins/dummy')),
+            Example2Plugin: toModulePlugin('Example2', () => import('../../test-resources/module-plugins/dummy2'))
+        };
+        const pluginsConfig = [
+            'ExamplePlugin',
+            'Example2Plugin'
+        ];
+
+        ReactDOM.render(<Component plugins={plugins} config={pluginsConfig} />, document.getElementById('container'));
+        setTimeout(() => {
+            ReactDOM.render(<Component
+                plugins={{ExamplePlugin: toModulePlugin('Example', () => import('../../test-resources/module-plugins/dummy'))}}
+                config={['ExamplePlugin']} />, document.getElementById('container'));
+        }, 200);
+        setTimeout(() => {
             done();
+        }, 200);
+    });
+    it('test epics should be muted if route has changed', done => {
+        const newStore = {
+            dispatch() { },
+            getState: () => {
+                return {};
+            },
+            subscribe() {
+            },
+            replaceReducer: () => { },
+            storeManager: {
+                reduce: () => {},
+                addReducer: () => {},
+                removeReducer: () => {},
+                addEpics: () => {},
+                muteEpics: (plugin) => {
+                    expect(plugin).toBe('Example2Plugin');
+                    done();
+                },
+                unmuteEpics: () => {},
+                rootEpic: () => {}
+            }
         };
-        ReactDOM.render(<Component onLoaded={assertCallback} />, document.getElementById('container'));
+        setStore(newStore);
+
+        const plugins = {
+            ExamplePlugin: toModulePlugin('Example', () => import('../../test-resources/module-plugins/dummy')),
+            Example2Plugin: toModulePlugin('Example2', () => import('../../test-resources/module-plugins/dummy2'))
+        };
+        const pluginsConfig = [
+            'ExamplePlugin',
+            'Example2Plugin'
+        ];
+
+        ReactDOM.render(<Component plugins={plugins} config={pluginsConfig} />, document.getElementById('container'));
+        setTimeout(() => {
+            window.history.pushState('page2', 'Title', '/mapstore/#/newPage');
+            ReactDOM.render(<Component
+                plugins={{ExamplePlugin: toModulePlugin('Example', () => import('../../test-resources/module-plugins/dummy'))}}
+                config={['ExamplePlugin']} />, document.getElementById('container'));
+        }, 200);
+        setTimeout(() => {
+            expect(true).toBe(false);
```

Comment:
so this is not failing because you call done before and the test was to make sure mute epics was called, right??

Reply:
Yes. Otherwise, if it doesn't go inside storeManager.muteEpics() - it will fail with timeout